src/Reveal.Sdk.Dom/Visualizations/Visualization.cs

```diff
@@ -106,6 +106,11 @@ public void UpdateDataSourceItem(DataSourceItem dataSourceItem)
             if (DataDefinition is TabularDataDefinition tdd)
             {
                 tdd.Fields = dataSourceItem.Fields.Clone();
+                
+                if(tdd.DataSourceItem.JoinTables != null)
+                {
+                    tdd.JoinTables.AddRange(tdd.DataSourceItem.JoinTables.Clone());

```

Comment:
this should be setting the JoinTables property instead of "AddRange".

Reply:
Changed to assignment